src/cpp/src/llm/pipeline_stateful.cpp

```diff
@@ -158,14 +158,6 @@ DecodedResults StatefulLLMPipeline::generate(
         }
     }
 
-    if (m_is_npu) {
-        // Prefill model in NPU is reshaped to NPUW_LLM_MAX_PROMPT_LEN x NPUW_LLM_MAX_PROMPT_LEN
-        OPENVINO_ASSERT(encoded_input.input_ids.get_size() <= m_max_prompt_len,
-            "Stateful LLM pipeline on NPU may only process prompts or hold chat history up to ",
-            m_max_prompt_len, " tokens. ", encoded_input.input_ids.get_size(), " is passed.\n"
-            "Set the \"MAX_PROMPT_LEN\" config option to increase the limit.");
-    }
```

Comment:
Should this really be removed?

Reply:
Yes, it is not needed anymore! This function call `generate()` for tokenized inputs, that was updated with the check in this PR!